### PR TITLE
Take input striding for conv forward based on eager output

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88891
* __->__ #88706
* #88534

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

From discussion with @Chillee and @ngimel we'll likely need further fixes to ensure that we hit channels last kernels but this is still worth landing in its own right.

### CODE REVIEW

**Comment by ngimel** [comment]
Can you also make FixedLayout instead of FlexibleLayout? That'll make cl propagation more robust. 
A time will come (soon) when we'll want to use cl in inductor even when eager was channels first, so we'll have to rethink the strategy of taking strides from eager, but for now to enable cl perf that's fine.

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -3016,9 +3016,27 @@ def create(
         output_padding_: List[int],
         groups: int,
     ):
+        with torch._subclasses.FakeTensorMode():
+            x_fake = ir_node_to_tensor(x, guard_shape=False)
+            weight_fake = ir_node_to_tensor(weight, guard_shape=False)
+            bias_fake = (
+                ir_node_to_tensor(bias, guard_shape=False) if bias is not None else bias
+            )
+            output = torch.ops.aten.convolution(
```
Comment (XiaobingSuper) [inline torch/_inductor/ir.py:3025]: If we add this path to query output strider order, do we still need the following path: 

````
  with FakeTensorMode():
            output, *_ = cls.process_kernel(
                torch.ops.aten.convolution,
                x,
                weight,
                bias,
                stride,
                padding,
                dilation,
                transposed,
                output_padding,
                groups,
            )
```

**Review by ngimel — APPROVED** [review]
The test is weak (it would have passed even without this PR), but that's fine.

**Comment by Chillee** [comment]
Don't we also need to do this for convolutionbackward?

**Comment by eellison** [comment]
> The test is weak (it would have passed even without this PR), but that's fine.

Not saying it's strong but prior we were taking striding for the weight

**Comment by ngimel** [comment]
Hm, before this PR cl input + contig weight did cl convolution

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -3016,32 +3024,39 @@ def create(
...
+                dilation_,
+                transposed,
+                output_padding_,
+                groups,
+            )
+            req_stride_order = get_stride_order(output.stride())
+
+        if config.triton.convolution == "aten":
+            weight = cls.require_stride_order(weight, req_stride_order)
+            x = cls.require_stride_order(x, req_stride_order)
```
Comment (jgong5) [inline torch/_inductor/ir.py:3048]: For CPU convolution, we would always convert the weight to the blocked layout inside the ATen conv. If the stride order of `weight` is different from output, there would be extra copy by requiring `weight` to follow the output stride. Shall we keep the weight stride intact for CPU backend here?
```suggestion
        if is_cpu():
            weight = cls.require_stride1(cls.realize_input(weight))
            x = cls.require_stride_order(x, req_stride_order)        
        elif config.triton.convolution == "aten":
            weight = cls.require_stride_order(weight, req_stride_order)
            x = cls.require_stride_order(x, req_stride_order)
```

**Pull request closed without merging**